session.go

```diff
@@ -714,13 +732,26 @@ func (q Query) String() string {
 
 //Attempts returns the number of times the query was executed.
 func (q *Query) Attempts() int {
-	return q.attempts
+	q.session.mu.Lock()
+	defer q.session.mu.Unlock()
```

Comment:
Why are we now locking the session mutex around the place?

Reply:
will change